website_chd_product_configurator/controllers/controllers.py

```diff
@@ -0,0 +1,244 @@
+                # selection in pricecomponent_name, the id encoded in the key
+                # if it is of type numerical the value is the actual value of
+                # the pricecomponent field, the id is encoded in the key
+                if ('pricecomponent_string' in key) or ('pricecomponent_int' in key):
+                    pricecomponent_id = int(key.split('_')[3])
+                    pricecomponent_value = form_data[key]
+                # the configurator stores the attributes in a
+                # dictionary in the "attributes" field ,
+                all_attributes[
+                    '_attribute_' + str(pricecomponent_id)
+                    ] = int(pricecomponent_value)
+
+        # add attributes to the configurator
+        chd_dict['attributes'] = str(all_attributes)
+        # dictionary is complete, create configurator
+        try:
+            new_chd = request.env['chd.product_configurator'].create(chd_dict)
+        except orm.except_orm as e:
+            errormsg = e.value
+        except AssertionError as a:
+            errormsg = a.value
+        # add accessories and price components selections to the configurator
+        list = []
+        for key in form_data:
+            # get only accessories that have been checked
+            if ('accessoryid_' in key) and form_data[key] != 0:
+                # extrapolate the id encoded the name
+                accessory_id = int(key.split('_')[1])
+                # get the associated value by choosing the field with the
+                # right name accessoryid_{id}=on/off and the associated
+                # quantity would be qtyaccessoryid_{id}=9898
+                accessory_qty = int(form_data[key])
+                list.append((0, 0, {
+                    'product_id': accessory_id,
+                    'configurator_id': new_chd.id,
+                    'quantity': accessory_qty, }
+                    ))
+        if list != []:
+            new_chd.write({'accessoire_line_ids':list})
+        # oduct configurator is ready, we can now calculate options
+        # model refers to old API model, self.pool
+        # is not available in controller context (praise the lord for Holger!)
+        try:
+            res = new_chd._model.calculate_price(
+                request.cr, request.uid, [new_chd.id], context=request.context)
+            results = chd_results_model.search([
+                ('configurator_id', '=', new_chd.id)])
+        except orm.except_orm as e:
+                errormsg += e.value
+
+        if errormsg != "":
+            product_type_model = request.env['product.type']
+            product_product_model = request.env['product.product']
+            curr_types = product_type_model.search([
+                ('product_option_ids', 'in', [int(form_data['product_id'])])
+                ])
+            avail_accessories = product_product_model.search(
+                [('id', 'in', curr_product_id.chd_accessoire_ids.ids)])
```

Comment:
that's the same as `avail_accessories = curr_product_id.chd_accessoire_ids`, you won't need product_product_model above then